modules/Bio/EnsEMBL/IO/Object/VCF4Metadata.pm

```diff
@@ -0,0 +1,178 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::IO::Object::VCF4Metadata - Object to represent VCF 4.2 metadata
+
+=head1 SYNOPSIS
+
+  use Bio::EnsEMBL::IO::Object::VCF4Metadata;
+
+  $record = Bio::EnsEMBL::IO::Object::VCF4Metadata->new($line);
+
+  $type = $record->{type};
+
+  $directive = $record->{directive};
+
+  @values = $record->{value};
+
+  $line = $record->create_record;
+
+=head1 Description
+
+An object to hold a generic column based format's record as a structure. Allows access to
+elements via setters/getters. Setter/getters are dynamically added to the class on instantiation.
+
+=cut
+
+package Bio::EnsEMBL::IO::Object::VCF4Metadata;
+
+use base qw/Bio::EnsEMBL::IO::Object::Metadata/;
+
+use strict;
+use warnings;
+use Carp;
+
+my $FORMAT_HEADER = 'FORMAT';
+my @HEADER = ('CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO');
+my @HEADER_FORMAT = @HEADER;
+push @HEADER_FORMAT, $FORMAT_HEADER;
+
+my %INFO = ( 'AA'  => '<ID=AA,Number=1,Type=String,Description="Ancestral Allele">',
+             'AC'  => '<ID=AC,Number=1,Type=Integer,Description="Allele Count">',
+             'AF'  => '<ID=AF,Number=A,Type=Float,Description="Allele Frequency">',
+             'AN'  => '<ID=AN,Number=1,Type=Integer,Description="Total Number of Alleles">',
+             'MA'  => '<ID=MA,Number=1,Type=String,Description="Minor Allele">',
+             'MAF' => '<ID=MAF,Number=1,Type=Float,Description="Minor Allele Frequency">',
+             'MAC' => '<ID=MAC,Number=1,Type=Integer,Description="Minor Allele Count">',
+             'NS'  => '<ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data">'
+           );
+
+my %FORMAT= ( 'GT' => '<ID=GT,Number=1,Type=String,Description="Genotype">' );
+
+sub new {
+    my ($class, $line) = @_;
+
+    my $self = {};
+
+    my ($type, @rest) = split /\s+/, $line;
+
+    if($type =~ /^##(\S+)/) {
+	    $self->{type} = 'directive';
+	    $self->{directive} = $1;
+	    $self->{value} = \@rest;
+    } elsif($type =~ /^#(\S+)/) {
+      my $indiv_col_start = length(@HEADER_FORMAT)-1;
+      my %column_header = map { $_ => 1 } @rest;
+      if (!$column_header{$FORMAT_HEADER}) {
+        $indiv_col_start = length(@HEADER)-1;
+      }
+      my @indiv = splice @rest, $indiv_col_start;
+	    $self->{type} = 'header';
+	    $self->{value} = \@indiv;
+    }
+
+    bless $self, $class;
+
+    return $self;
+}
+
+=head2 directive
+
+    Description: Create a directive type VCF metadata (##)
+    Args[1]    : The directive type (ie. fileformat)
+    Args[2]    : Directive values, either a string or array of
+                 values. An array will be put together as a '='
+                 separated string when create_record is called.
+    Returntype : Bio::EnsEMBL::IO::Object::VCF4Metadata
+
+=cut
+
+sub directive {
+    my $class = shift;
+    my $directive = shift;
+    my @args = @_;
```

Comment:
You should provide an example for this, given the open-ended @args isn't in the Args docs above.